File: libraries/cyclestreets-core/src/main/java/net/cyclestreets/api/client/ApiKeyInterceptor.java
package net.cyclestreets.api.client;

import java.io.IOException;

import okhttp3.HttpUrl;
import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

public class ApiKeyInterceptor implements Interceptor {

  private final String apiKey;

  public ApiKeyInterceptor(String apiKey) {
    this.apiKey = apiKey;
  }

  @Override
  public Response intercept(Chain chain) throws IOException {
    Request request = chain.request();
    HttpUrl newUrl = request.url().newBuilder().addQueryParameter("key", apiKey).build();
    Request newRequest = request.newBuilder().url(newUrl).build();
    return chain.proceed(newRequest);
  }
}

File: libraries/cyclestreets-core/src/main/java/net/cyclestreets/api/client/ApiKeyInterceptor.kt
package net.cyclestreets.api.client

import java.io.IOException

import okhttp3.Interceptor
import okhttp3.Response

class ApiKeyInterceptor(private val apiKey: String) : Interceptor {

    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        val request = chain.request()
        val newUrl = request.url().newBuilder().addQueryParameter("key", apiKey).build()
        val newRequest = request.newBuilder().url(newUrl).build()
        return chain.proceed(newRequest)
    }
}
